jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -199,7 +200,7 @@ public static Request getBaseRequest(ServletRequest request)
     private boolean _handled = false;
     private boolean _contentParamsExtracted;
     private boolean _requestedSessionIdFromCookie = false;
-    private Attributes _attributes;
+    private Attributes _attributes = new ServletAttributes();
```

Comment:
Ah we can't make this final because of possible wrappers that don't unwrap.
So go back to lazy creation of this field......  OR have 2 fields:
```suggestion
    private final AsyncAttributesMap _asyncAttributes = new AsyncAttributesMap();
    private Attributes _attributes = _asyncAttributes;
```

This way, async could as the request directly for the `_asyncAttributes` and set the fields on it.... but we'd still create an extra class with 6 fields even if we never use attributes.... hmmmm @lorban your thoughts on which is best?

Reply:
I would leave the `_attributes` field non-final, this is the simplest way to go.

Any sort of cleanup seem to require a non-negligible amount of refactoring, so I'd keep #4861 open until the code is cleaned up. Regarding perf, this small change has 99% of the benefits of a larger refactoring.